modules/compute/gke-node-pool/variables.tf

```diff
@@ -265,3 +265,29 @@ variable "service_account" {
     error_message = "service_account is deprecated and replaced with service_account_email and scopes."
   }
 }
+
+variable "reservation_type" {
+  description = "Type of reservation to consume"
```

Comment:
Can we add here how to create this reservation ? And also mention that we assume this is a valid reservation ? 

Also what happens when this is not a valid reservation, does the GKE cluster creation throws error ? And is that error visible to the customer ?

Reply:
> Can we add here how to create this reservation ? And also mention that we assume this is a valid reservation ?

Done. Updated the specific_reservation's description. I think, it's more relevant there.

> Also what happens when this is not a valid reservation, does the GKE cluster creation throws error ? And is that error visible to the customer ?

If it is not a valid reservation or the reservation does not have enough capacity, the cluster (and node pool) creation succeeds with problems**  which are not apparent in the Toolkit's deploy logs. 

**: Node pool creation succeeds but the GCE Instance creation fails in both of the mentioned cases. Failure reason is available in the node pool's status. I am not sure whether it's correct behaviour. We can check this with the GKE team if this is working as designed or a bug -- node pool creation succeeded but the nodes backing the pool cannot be created.

As far as Toolkit is concerned, do you think, the variable description asking to ensure the name and capacity of the reservation is sufficient? Ideally, the error should be visible in the Toolkit's logs. I will explore how to modify Toolkit's logs to use the node pool's status (it is available in the node-pools describe API)

I should also mention that as soon as a matching reservation is created or additional capacity is added (offline), the GCE Instance creation succeeds.

Reply:
Thanks for checking, let's add a task to throw an error if the reservation is not found at the time of "blueprint deployment" to unblock this reservation.